src/main/java/io/dropwizard/kafka/managed/KafkaAdminClientManager.java

```diff
@@ -17,16 +18,27 @@
     private final String name;
     private final Collection<NewTopic> topics;
 
-    public KafkaAdminClientManager(final AdminClient adminClient, final String name, Collection<NewTopic> topics) {
+    public KafkaAdminClientManager(final AdminClient adminClient, final String name, final Collection<NewTopic> topics) {
         this.adminClient = requireNonNull(adminClient);
         this.name = requireNonNull(name);
         this.topics = topics;
     }
 
     @Override
     public void start() throws Exception {
-        if (this.topics != null && !this.topics.isEmpty()) {
-            // TODO: Check if topics already exist
+        log.info("Starting adminClient for name={}", name);
+        if (!this.topics.isEmpty()) {
+            log.trace("Searching existing topics in cluster.");
+            final Set<String> existingTopics = this.adminClient.listTopics().names().get();
+            for (String t : existingTopics) {
+                this.topics.removeIf(newTopic -> {
+                    boolean match = newTopic.name().equals(t);
+                    if (match) {
+                        log.warn("Not attempting to re-create existing topic {}.", newTopic.name());
```

Comment:
I think maybe a debug log (or a single info log with all of the skipper topics appended together) would be more appropriate. Otherwise, every time someone restarts their app, they may be flooded with warn logs.